catalog/spatial/geocoding/spatial-geocoding-feature/src/main/java/org/codice/ddf/spatial/geocoding/feature/CatalogFeatureIndexer.java

```diff
@@ -0,0 +1,179 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.geocoding.feature;
+
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.io.WKTWriter;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.security.service.SecurityServiceException;
+import java.io.Serializable;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.codice.ddf.security.common.Security;
+import org.codice.ddf.spatial.geocoding.FeatureExtractionException;
+import org.codice.ddf.spatial.geocoding.FeatureExtractor;
+import org.codice.ddf.spatial.geocoding.FeatureIndexer;
+import org.codice.ddf.spatial.geocoding.FeatureIndexingException;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.filter.Filter;
+
+public class CatalogFeatureIndexer implements FeatureIndexer {
+
+  private Security security = Security.getInstance();
+
+  private CatalogFramework catalogFramework;
+
+  private FilterBuilder filterBuilder;
+
+  public void setSecurity(Security security) {
+    this.security = security;
+  }
+
+  public void setCatalogFramework(CatalogFramework catalogFramework) {
+    this.catalogFramework = catalogFramework;
+  }
+
+  public void setFilterBuilder(FilterBuilder filterBuilder) {
+    this.filterBuilder = filterBuilder;
+  }
+
+  @Override
+  public void updateIndex(
+      String resource, FeatureExtractor featureExtractor, boolean create, IndexCallback callback)
+      throws FeatureExtractionException, FeatureIndexingException {
+
+    if (create) {
+      removeExistingMetacards();
+    }
+
+    AtomicInteger count = new AtomicInteger(0);
```

Comment:
Why `AtomicInteger`?

Reply:
Looked like the quickest way to have the lambda increment a counter outside of its scope. Is there more self-explanatory way to do this?

Reply:
Just curious; seems fine to me

Reply:
Would it be possible to use a method reference here instead of a lamba?
You could then get around the fact that lambdas require anything they touch to be effectively final.

Reply:
Sounds more self-explanatory to me :+1: